src/views/AddItem.jsx

```diff
@@ -1,3 +1,65 @@
+import { useState } from 'react';
+import { addItem } from '../api/firebase.js';
+
 export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+	const [itemName, setItemName] = useState('');
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(7);
+	const [itemAdded, setItemAdded] = useState(false);
+	const [error, setError] = useState(false);
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		try {
+			await addItem('my test list', {
+				itemName,
+				daysUntilNextPurchase,
+			});
+			setItemAdded(true);
+			setError(false);
+		} catch (e) {
+			setError(true);
+			setItemAdded(false);
+		}
+	};
+
+	return (
+		<>
+			<form onSubmit={handleSubmit}>
+				<label htmlFor="item-name-input">Item name:</label>
+				<input
+					type="text"
+					id="item-name-input"
+					onChange={(e) => setItemName(e.target.value)}
+				/>
+				<fieldset>
+					<legend>How soon will you buy this again?</legend>
+					<input
+						type="radio"
+						id="soon"
+						onChange={() => setDaysUntilNextPurchase(7)}
+						checked={daysUntilNextPurchase === 7}
+					/>
+					<label htmlFor="soon">Soon</label> <br />
+					<input
+						type="radio"
+						id="kind-of-soon"
+						onChange={() => setDaysUntilNextPurchase(14)}
+						checked={daysUntilNextPurchase === 14}
+					/>
+					<label htmlFor="kind-of-soon">Kind of soon</label> <br />
+					<input
+						type="radio"
+						id="not-soon"
+						onChange={() => setDaysUntilNextPurchase(30)}
```

Comment:
To ["DRY up your code"](https://www.digitalocean.com/community/tutorials/what-is-dry-development), you could save these numbers to local variables in your component and then reference those variables in your props instead of explicitly referencing the numbers. I believe the arguments are that it can help with clarity, it clusters these definitions which makes it easier to update them the future, and reduces the possibility of introducing errors in the application (ex: if in the future the team decides that 'soon' should be `5` days instead of `7` days and a dev updates the `onChange` prop but forgets to update the respective `checked` prop). 

Super trivial and not necessary here but I thought I'd include it in case it's helpful in the future.

Reply:
I think that is really helpful, Jon! Thanks for the input!

Reply:
There are some good (but not urgent!) opportunities to DRY this up! We don't have to do any of that now, but we probably _will_ want to, for maintainability, as Jon mentioned. In addition to Jon's specific idea, consider: in the code that renders your UI, are there very repetitive parts with little change? How can you consolidate those parts into some repeatable code – some kind of loop? 🤔

Reply:
Thanks for that feedback, @jongranados and @mxmason! That's great food for thought.